src/app/src/main/kotlin/org/imperial/mrc/hint/db/CalibrateDataRepository.kt

```diff
@@ -51,11 +58,50 @@ class JooqCalibrateDataRepository: CalibrateDataRepository
         }
     }
 
-    private fun getDataFromConnection(conn: Connection): List<CalibrateResultRow> {
-        val query = DEFAULT_QUERY
+    private fun convertIndicatorToList(resultSet: ResultSet): List<String> {
+        resultSet.use {
+            return generateSequence {
+                if (it.next()) {
+                    it.getString("indicator")
+                } else {
+                    null
+                }
+            }.toList()
+        }
+    }
+
+    private fun getIndicatorQuery(indicator: String): String {
+        return """SELECT
+        age_group,area_id,
+        calendar_quarter,
+        indicator,
+        ROUND(lower, 4) AS lower,
+        ROUND(mean, 4) AS mean,
+        ROUND(mode, 4) AS mode,
+        sex,
+        ROUND(upper, 4) AS upper
+        FROM data WHERE indicator='$indicator'"""
+    }
+
+    private fun getDataFromConnection(
+        conn: Connection,
+        indicator: String): List<CalibrateResultRow> {
+        val query: String
+        if (indicator == "all") {
+            query = DEFAULT_QUERY
+        } else {
+            val indicatorStmt: Statement = conn.createStatement()
+            val indicatorResultSet = indicatorStmt.executeQuery(INDICATOR_QUERY)
+            val validIndicators = convertIndicatorToList(indicatorResultSet)
```

Comment:
I wonder if this is necessary if we're using prepared statement like above? Also feels like we query for an indicator which doesn't exist we should just return 0 rows, the actual query was valid and fine but just no data available? 

Does returning the error make it easier to handle in the front end though?

Reply:
No you're right, let's do it with a prepared statement and if it doesn't match we should at least return an empty array, I guess it's not really an error if the indicator doesn't exist but equally this may make our lives difficult in the future when we try and debug why data may not be showing up on plots, I wonder if we should error and the frontend should handle it if it's an "indicator not present" error?

Reply:
after more thought on this, it is probably best to not distinguish the two and just return an empty array to the frontend!